engine/src/it/java/org/pentaho/di/job/JobExportIT.java

```diff
@@ -0,0 +1,180 @@
+
+  private static final String REPOSITORY_ROOT_DIR = "src/it/resources/org/pentaho/di/job/repo";
+  private static final String REPOSITORY_DIR = "/folder";
+  private static final String JOB_NAME = "sample_job";
+  private static final String EXTRACT_DIR = EXPORT_DIR + File.separator + JOB_NAME;
+  private static final String EXTRACTED_JOB_FILE = EXTRACT_DIR + File.separator + JOB_NAME + ".kjb";
+  private static final String EXTRACTED_TRANS_FILE = EXTRACT_DIR + File.separator + "sample_trans.ktr";
+
+  private KettleFileRepository repository;
+
+  @BeforeClass
+  public static void setUpOnce() throws KettleException {
+    deleteFolder( new File( EXPORT_DIR ) );
+    new File( EXPORT_DIR ).mkdirs();
+
+    KettleEnvironment.init();
+  }
+
+  @AfterClass
+  public static void tearDownOnce() {
+    deleteFolder( new File( EXPORT_DIR ) );
+  }
+
+  @Before
+  public void setUp() throws KettleException {
+    KettleFileRepositoryMeta repositoryMeta = new KettleFileRepositoryMeta( FILE_REPOSITORY_ID, FILE_REPOSITORY_NAME,
+      FILE_REPOSITORY_DESC, REPOSITORY_ROOT_DIR );
+    repository = new KettleFileRepository();
+    repository.init( repositoryMeta );
+    repository.connect( null, null );
+  }
+
+  @After
+  public void tearDown() {
+    repository.disconnect();
+  }
+
+  /**
+   * Given a Job located in non-root directory of a repository,
+   * and referencing to a Transformation using 'Internal.Entry.Current.Directory' variable.
+   * <br/>
+   * When this Job is exported into a zip file,
+   * then the referenced Transformation should be exported as well.
+   */
+  @Test
+  public void shouldExportJobAndRelatedTransformationFile() throws IOException, MetaStoreException, KettleException {
+    RepositoryDirectoryInterface repositoryDir = repository.loadRepositoryDirectoryTree();
+    repositoryDir = repositoryDir.findDirectory( REPOSITORY_DIR );
+
+    JobMeta jobMeta = repository.loadJob( JOB_NAME, repositoryDir, null, null );
+    Job job = new Job( repository, jobMeta );
+
+    ResourceUtil.serializeResourceExportInterface( EXPORT_FILE, job.getJobMeta(), job, repository, null );
+
+    File zipFile = new File( EXPORT_FILE );
+    assertTrue( zipFile.exists() );
+
+    extractZip( zipFile, EXTRACT_DIR );
+
+    // assert that either of files, job and transformation, have been exported.
+    assertTrue( new File( EXTRACTED_JOB_FILE ).exists() );
+    assertTrue( new File( EXTRACTED_TRANS_FILE ).exists() );
+  }
+
+  private static void deleteFolder( File folder ) {
+    File[] files = folder.listFiles();
+    if ( files != null ) { //some JVMs return null for empty dirs
+      for ( File f : files ) {
+        if ( f.isDirectory() ) {
+          deleteFolder( f );
+        } else {
+          f.delete();
+        }
+      }
+    }
+    folder.delete();
+  }
+
+  private static void extractZip( File zipFile, String extractDir ) throws IOException {
+    int BUFFER = 2048;
+    ZipFile zip = new ZipFile( zipFile );
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *zip*. Execution paths that do not contain closure statements were detected. Ensure *zip* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).